src/Components /SingleArticle.jsx

```diff
@@ -1,28 +1,54 @@
-import { useParams } from "react-router-dom";
-import { useEffect, useState } from "react";
-import * as api from "../api.js";
-import CommentsList from "./ArchiveCommentsList.jsx";
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import * as api from '../api.js';
+import CommentsList from './ArchiveCommentsList.jsx';
+import ArticleRating from './ArticleVotes.jsx';
 
 export default function SingleArticle() {
-  const {article_id} = useParams();
-  const [singleArticle, setSingleArticle] = useState({});
+	const [singleArticle, setSingleArticle] = useState({});
+	const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false)
+	
+	const { article_id } = useParams();
 
-useEffect(() => {
-  api.fetchSingleArticle(article_id).then((data) => {
-    setSingleArticle(data.article)
-  })
-}, []);
-console.log(singleArticle)
+	useEffect(() => {
+		setIsLoading(true);
+		api.fetchSingleArticle(article_id)
+			.then((data) => {
+				setSingleArticle(data.article);
+			})
+			.catch((err) => {
+				setIsError(true)
+      })
+      .finally (() => {
+        setIsLoading(false)
+			});
+	}, [article_id]);
 
-return (
-	<div>
-		<h1>{singleArticle.topic}</h1>
 
-		<h1>{singleArticle.title}</h1>
-		<h2>{singleArticle.author}</h2>
-		<h2>{singleArticle.body}</h2>
-		<CommentsList articleId={article_id} />
-	</div>
-);
 
-}
\ No newline at end of file
+	if (isLoading) return <p className="loading">Loading...</p>;
+	
+  if (isError) return <p> something went wrong, please try again later ...</p>;
+
+	return (
+		<div>
+			<img
+				className="article-img"
+				src={singleArticle.article_img_url}
+				alt={`${singleArticle.topic}`}
+			/>
+			<h1>{singleArticle.topic}</h1>
+			<h1>{singleArticle.title}</h1>
```

Comment:
Ideally you only have ONE h1 tag on a page, but any header tags after that e.g. h2, h3.. can have multiple instances